code/game/machinery/computer/security.dm

```diff
@@ -2,311 +2,192 @@
+	var/datum/data/record/record_security = null
+	/// Whether the computer is currently printing a paper or not.
+	var/is_printing = FALSE
+	/// The editable fields and their associated question to display to the user.
+	var/static/list/field_edit_questions
+	/// The editable fields and their associated choices to display to the user.
+	var/static/list/field_edit_choices
+	/// The current temporary notice.
+	var/temp_notice
 
 	light_color = LIGHT_COLOR_RED
 
+/obj/machinery/computer/secure_data/Initialize(mapload)
+	. = ..()
+	req_one_access = list(ACCESS_SECURITY, ACCESS_FORENSICS_LOCKERS)
+	field_edit_questions = list(
+		// General
+		"name" = "Please input new name:",
+		"id" = "Please input new ID:",
+		"sex" = "Please select new sex:",
+		"age" = "Please input new age:",
+		"rank" = "Please input new assignment:",
+		"fingerprint" = "Please input new fingerprint hash:",
+		"p_stat" = "Please select new physical status:",
+		"m_stat" = "Please select new mental status:",
+		// Security
+		"criminal" = "Please select new criminal status:",
+		"mi_crim" = "Please input new minor crimes:",
+		"mi_crim_d" = "Please summarize minor crimes:",
+		"ma_crim" = "Please input new major crimes:",
+		"ma_crim_d" = "Please summarize major crimes:",
+		"notes" = "Please input new important notes:",
+	)
+	field_edit_choices = list(
+		// General
+		"sex" = list("Male", "Female"),
+		"p_stat" = list("*Deceased*", "*SSD*", "Active", "Physically Unfit", "Disabled"),
+		"m_stat" = list("*Insane*", "*Unstable*", "*Watch*", "Stable"),
+		// Security
+		"criminal" = list(SEC_RECORD_STATUS_NONE, SEC_RECORD_STATUS_ARREST, SEC_RECORD_STATUS_EXECUTE, SEC_RECORD_STATUS_INCARCERATED, SEC_RECORD_STATUS_RELEASED, SEC_RECORD_STATUS_PAROLLED, SEC_RECORD_STATUS_DEMOTE, SEC_RECORD_STATUS_SEARCH, SEC_RECORD_STATUS_MONITOR),
+	)
+
 /obj/machinery/computer/secure_data/Destroy()
-	active1 = null
-	active2 = null
+	record_general = null
+	record_security = null
 	return ..()
 
 /obj/machinery/computer/secure_data/attackby(obj/item/O, mob/user, params)
-	if(istype(O, /obj/item/card/id) && !scan)
-		user.drop_item()
-		O.forceMove(src)
-		scan = O
-		ui_interact(user)
+	if(tgui_login_attackby(O, user))
 		return
 	return ..()
 
-//Someone needs to break down the dat += into chunks instead of long ass lines.
 /obj/machinery/computer/secure_data/attack_hand(mob/user)
 	if(..())
 		return
 	if(is_away_level(z))
 		to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the station!")
 		return
 	add_fingerprint(user)
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/computer/secure_data/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/computer/secure_data/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
```

Comment:
You change force_open=1 to force_open=FALSE

Reply:
Whoops, well spotted!